mario/tools/tableparser.py

```diff
@@ -1706,3 +1709,1059 @@ def parser_figaro_e3(path,doping_value):
+            df_filled['row_name']=row_name_categ+'_'+df_filled[row_name_reg]+'_'+df_filled['COMM']
+            df_filled.drop(columns='SRC',inplace=True)
+            df_filled.drop(columns='COMM',inplace=True)
+        elif row_name_setting=='only_comm':
+            df_filled['row_name']=row_name_categ+'_REG_'+df_filled['COMM']
+        elif row_name_setting=='only_categ':
+            df_filled['row_name']=row_name_categ+'_REG'
+        elif row_name_setting=='emi_dom':
+            df_filled['row_name']=row_name_categ+'_'+df_filled['EM']+'_dms_'+df_filled['COMM'] #careful: 'EM' element has different lengths
+            df_filled.drop(columns='EM',inplace=True)
+            df_filled.drop(columns='SRC',inplace=True)
+            df_filled.drop(columns='COMM',inplace=True)
+        elif row_name_setting=='emi_imp':
+            df_filled['row_name']=row_name_categ+'_'+df_filled['EM']+'_'+df_filled['SRC']+'_'+df_filled['COMM']
+            df_filled.drop(columns='EM',inplace=True)
+            df_filled.drop(columns='SRC',inplace=True)
+            df_filled.drop(columns='COMM',inplace=True)
+        elif row_name_setting=='ene_dom':
+            df_filled['row_name']=row_name_categ+'_dms_'+df_filled['COMM'] #domestic
+            df_filled.drop(columns='SRC',inplace=True)
+            df_filled.drop(columns='COMM',inplace=True)
+        elif row_name_setting=='ene_imp':
+            df_filled['row_name']=row_name_categ+'_'+df_filled['SRC']+'_'+df_filled['COMM']
+            df_filled.drop(columns='SRC',inplace=True)
+            df_filled.drop(columns='COMM',inplace=True)
+
+
+        # 5) If need to split_agent
+        if split_agent:
+            if 'AGENT' in df_filled.columns:
+                df_V= df_filled[df_filled['AGENT'].isin(indeces['s']['main'])]
+                df_VY = df_filled[df_filled['AGENT'].isin(indeces['n']['main'])]
+            elif 'AGT' in df_filled.columns:
+                df_V = df_filled[df_filled['AGT'].isin(indeces['s']['main'])]
+                df_VY = df_filled[df_filled['AGT'].isin(indeces['n']['main'])]
+
+            print('Starting pivoting now')
+            # Pivot di part1
+            pivot_V = df_V.pivot_table(
+                index=pivot_index, 
+                columns=pivot_columns, 
+                values='VALUE', 
+                aggfunc='sum'
+            ).fillna(0)
+
+            # Pivot di part2
+            pivot_VY = df_VY.pivot_table(
+                index=pivot_index, 
+                columns=pivot_columns, 
+                values='VALUE', 
+                aggfunc='sum'
+            ).fillna(0)
+
+            return pivot_V, pivot_VY
+
+        else:
+            # Pivot unico
+            matrix = df_filled.pivot_table(
+                index=pivot_index, 
+                columns=pivot_columns, 
+                values='VALUE', 
+                aggfunc='sum'
+            ).fillna(0)
+
+            return matrix
+
+    # Matrix Z and Y
+    print('Starting Z and Y')
```

Comment:
please avoid direct printing. In case you need to print something it needs to be done visa log_time function